template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidator.java

```diff
@@ -65,6 +68,21 @@ public void validateHostGroupScalingRequest(String accountId, Blueprint blueprin
         }
     }
 
+    private void validateNodeManagerAdjustment(HostGroup hostGroup, Integer adjustment, Collection<InstanceGroup> instanceGroups,
+            CmTemplateProcessor templateProcessor) {
+        Map<String, Set<String>> nodeManagerComponentInGroups = templateProcessor.getComponentsByHostGroup().entrySet().stream()
+                .filter(entry -> entry.getValue().contains("NODEMANAGER"))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        if (nodeManagerComponentInGroups.containsKey(hostGroup.getName())) {
+            instanceGroups.stream().filter(instanceGroup -> nodeManagerComponentInGroups.containsKey(instanceGroup.getGroupName()))
+                    .map(InstanceGroup::getNodeCount).reduce(Integer::sum).ifPresent(nodeManagerCountInGroups -> {
+                if (nodeManagerCountInGroups + adjustment <= 0) {
+                    throw new BadRequestException("At lease one instance required with NODEMANAGER role.");
```

Comment:
Please fix the typo in the second word! :)
Maybe one sentence could also be added to be more customer-friendly like `Your request would eliminate all the instances with NODEMANAGER role which is not supported. At least.....`

Reply:
@biharitomi I add another option below please check it.